### PR TITLE
Add rules: scss-function-pattern, scss-mixin-pattern

### PR DESCRIPTION
**Pull request by dryoma**
Implements #12

### CODE REVIEW

File: src/rules/scss-function-pattern/index.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      possible: [ isRegExp, isString ],
+    })
+    if (!validOptions) { return }
+
+    const regexpPattern = (isString(pattern))
+      ? new RegExp(pattern)
+      : pattern
+
+    root.walkAtRules(declaration => {
+      const decl = declaration
```
Comment (kristerkari) [inline src/rules/scss-function-pattern/index.js]: I think that the code would be a bit more readable if you could just change `declaration` to `decl` and do something like this:

```js
root.walkAtRules(decl => {
  if (decl.name !== "function") { return }

  const funcName = decl.params.replace(/(\s*?)\(.*?\)/g, "")
  if (regexpPattern.test(funcName)) { return }
```

Comment (dryoma) [inline src/rules/scss-function-pattern/index.js]: Good point, changed that part. In the code for these rules I relied on stylelint `-pattern` rules code a lot, so decided to leave that mostly unchanged.

**Comment by kristerkari** [comment]
I merged another PR and now there is a conflict. Could you rebase your branch against master?

**Comment by dryoma** [comment]
Sure, see #17

**Comment by kristerkari** [comment]
Closing this, let's continue with #18

**Pull request closed without merging**